File: backend/WebTestToolkit.Api/Hubs/PingHub.cs
using Microsoft.AspNetCore.SignalR;

namespace WebTestToolkit.Api.Hubs;

// Minimal round-trip proof that SignalR is wired end to end before any real hub
// (inspector/run progress) exists. Safe to delete once those land.
public class PingHub : Hub
{
    public Task<string> Ping() => Task.FromResult("pong");
}

File: frontend/src/api/pingHub.ts
import * as signalR from '@microsoft/signalr'

// Minimal SignalR round-trip against the placeholder PingHub, proving the live-event
// pipeline works end to end before any real hub (inspector/run progress) exists.
export async function pingHub(): Promise<string> {
  const connection = new signalR.HubConnectionBuilder()
    .withUrl('/hubs/ping')
    .build()

  try {
    await connection.start()
    return await connection.invoke<string>('Ping')
  } finally {
    await connection.stop()
  }
}
